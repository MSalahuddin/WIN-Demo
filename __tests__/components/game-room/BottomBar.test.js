import React from 'react';
import { render } from 'react-native-testing-library';

import BottomNavigator from '../../../js/components/game-room/BottomBar';
import MockProvider from '../../MockProvider';

describe('Component BottomNavigator', () => {
  const onPiggyBankPress = jest.fn();
  const navigation = { replace: jest.fn(), navigate: jest.fn(), goBack: jest.fn(), dispatch: jest.fn() };
  test('renders regular BottomNavigator correctly', () => {
    const output = render(
      <MockProvider>
        <BottomNavigator
         navigation={navigation}
         onPiggyBankPress= {onPiggyBankPress}
        />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

 
});
