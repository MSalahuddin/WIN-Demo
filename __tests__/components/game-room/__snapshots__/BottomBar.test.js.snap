// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component BottomNavigator renders regular BottomNavigator correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 10,
        "marginTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 12,
          "left": 12,
          "right": 12,
          "top": 12,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "opacity": 1,
        }
      }
      testID="icon-button"
    >
      <Image
        opacity={1}
        resizeMode="contain"
        size={45}
        source={1}
        style={
          Array [
            Object {
              "height": 67.5,
              "width": 67.5,
            },
          ]
        }
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
